docs/DeveloperGuide.adoc

```diff
@@ -364,6 +365,136 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+The commands are matched using the `contain()` function and comparing the first character of the user's input. Commands that are matched are placed in a list and mapped to their corresponding `HelperCard` classes.
+For example, an "add" command will create a new instance of `HelperCard` using "add" as a parameter for the `HelperCard` constructor.
+
+To emulate the behaviour of command helpers in IDEs, the use can also navigate the list of commands using the arrow keys. To prevent it from overwriting the existing code of loading input history, an if-else statement is used. The command box helper takes priority in this case; as long as there is a command
+that corresponds to the user's entered string in the CLI, i.e. the command box helper is shown, interaction with the command helper will take precedence over interactions with the CLI. This is implemented similarly in the key inputs `TAB` and `ENTER`, which are used to fill in the CLI after the user has selected his
+desired command in the command helper.
+
+==== Design Considerations
+
+**Aspect:** Implementation of the list view in `CommandBoxHelper` +
+**Alternative:** Instead of creating a new class to populate the ListView, use a StackPane instead of a ListView and add Strings to it +
+**Pros:** Easier to implement the display for command helper +
+**Cons:** Fragile architecture, have to change a few functions every time an extra command is added. This involves hardcoding the additional command Strings into the functions.
+Slightly harder to display selection of command in the helper box to the user.
+
+// tag::preferences[]
+=== Preferences
+
+In addition to configuration, user can customize the address book application by editing preferences. The difference between
+configurations and preferences is that configuration stores basic information requried to start the application, while
+preferences determine high level functionalities and UI styles. +
+
+==== Implementation
+The preferences are stored in `preferences.json` by default. This location is set in `config.json`. When the application
+is started, the `initPrefs` method reads the preferences file and converts it into a `UserPrefs` object using jackson-databind
+API. +
+In `UserPrefs` class, each valid preference key is defined as a field in the class, with initial value set to the default
+value of the preference. For example, the default path to the address book local storage is :
+[source,java]
+----
+private String addressBookFilePath = "data/addressbook.xml";
+----
+There are also two methods defined for each preference: `get<PreferenceKey>` and `set<PreferenceKey>`. These are used for
+reading and writing to the preference respectively. For example, for AddressBookFilePath, we have :
+[source,java]
+----
+public String getAddressBookFilePath() {
+    return addressBookFilePath;
+}
+
+public void setAddressBookFilePath(String addressBookFilePath) {
+    this.addressBookFilePath = addressBookFilePath;
+}
+----
```

Comment:
Code snippets are good, but try to keep them minimal, as you will need to update these if your codes change as well.